app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -45,10 +45,8 @@ def authenticate_user
 
   def handle_second_factor_locked_user(type)
     analytics.track_event(Analytics::MULTI_FACTOR_AUTH_MAX_ATTEMPTS)
-
-    sign_out
-
     render 'two_factor_authentication/shared/max_login_attempts_reached', locals: { type: type }
+    sign_out
```

Comment:
NOTE this order change -- I needed to do this, otherwise was getting a nil error on `current_user.decorate` -- I think it was working before because we memoized `decorated_user`